.github/workflows/create-release.yaml

```diff
@@ -0,0 +1,116 @@
+          CURRENT_RELEASE_TAG: ${{ github.event.inputs.tagName }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: ./scripts/release/get_release_by_tag.sh $CURRENT_RELEASE_TAG
+      - name: Get last release version
+        id: get_last_release_tag
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set +e
+          SCRIPT_OUTPUT=$(./scripts/release/get_last_release.sh)
+          SCRIPT_EXIT_CODE=$?
+          if [[ $SCRIPT_EXIT_CODE != 0 ]]; then
+            echo "$SCRIPT_OUTPUT"
+            exit $SCRIPT_EXIT_CODE
+          fi
+          set -e
+          echo "Last Release version: $SCRIPT_OUTPUT"
+          echo "last_release_tag=$SCRIPT_OUTPUT" >> $GITHUB_OUTPUT
+  create-tag:
+    name: Create lightweight tag
+    runs-on: ubuntu-latest
+    needs: validate-release
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Create lightweight tag
+        run: |
+          git tag ${{ github.event.inputs.tagName }} ${{ github.event.inputs.commitHash }}
+          git push origin ${{ github.event.inputs.tagName }}
+      - name: Wait for the Docker image
+        timeout-minutes: 20
+        env:
+          ITERATIONS: 40
+          SLEEP_SECONDS: 30
+        run: ./scripts/release/wait_for_image.sh ${{ env.IMAGE_REPO }}:${{ github.event.inputs.tagName }} $ITERATIONS $SLEEP_SECONDS
+  draft-release:
+    name: Create draft release
+    runs-on: ubuntu-latest
+    needs: [validate-release, create-tag]
+    outputs:
+      release_id: ${{ steps.draft_release.outputs.release_id }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Generate changelog
+        id: generate_changelog
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          CURRENT_RELEASE_TAG: ${{ github.event.inputs.tagName }}
+          LAST_RELEASE_TAG: ${{ needs.validate-release.outputs.last_release_tag }}
+          DOCKER_IMAGE_URL: ${{ env.IMAGE_REPO }}:${{ github.event.inputs.tagName }}
+        run: |
+          echo "Generating changelog for version: ${{ needs.validate-release.outputs.current_release_tag }}" 
+          CHANGELOG_FILE_NAME=$(./scripts/release/create_changelog.sh $CURRENT_RELEASE_TAG $LAST_RELEASE_TAG $DOCKER_IMAGE_URL)
+          echo "changelog_file_name=$CHANGELOG_FILE_NAME" >> $GITHUB_OUTPUT
+      - name: Create draft release
+        id: draft_release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          CURRENT_RELEASE_TAG: ${{ github.event.inputs.tagName }}
+          CHANGELOG_FILE_NAME: ${{ steps.generate_changelog.outputs.changelog_file_name }}
+        run: |
+          RELEASE_ID=$(./scripts/release/create_draft_release.sh $CURRENT_RELEASE_TAG $CHANGELOG_FILE_NAME)
+          echo "RELEASE_ID=$RELEASE_ID"
+          echo "release_id=$RELEASE_ID" >> $GITHUB_OUTPUT
+  publish_release:
```

Comment:
Just as an idea, how about we include a checkbox in the dispatch inbox for `publish: true/false`? By setting false we would have a chance to manually check the created entry before eventually publishing?

Reply:
I will add this option if you test if it works correctly :)